### DISCUSSION TITLE
Methods for developing new database dialects

### DISCUSSION THREAD

**Discussion by Lucaszlei** [sqlalchemy/sqlalchemy#7328] [Usage Questions] [answered]
I want to develop a new database dialect how do I do that？
Have relevant data to be able to consult below

**Comment by CaselIT** [comment, marked as answer]
Hi,

I would suggest starting from the readme: https://github.com/sqlalchemy/sqlalchemy/blob/main/README.dialects.rst
Then you can use the internal dialects https://github.com/sqlalchemy/sqlalchemy/tree/main/lib/sqlalchemy/dialects or external ones, like https://github.com/gordthompson/sqlalchemy-access or https://github.com/cockroachdb/sqlalchemy-cockroachdb as examples

**Reply by Lucaszlei** [reply]
thank you

**Reply by Abdur-rahmaanJ** [reply]
You can mark the issue as answered.
